File: java-hfsframework-primefaces/hfsframework-admin/src/main/java/br/com/hfsframework/admin/data/AdmMenuRepository.java
/**
 * <p><b>HFS Framework</b></p>
 * @author Henrique Figueiredo de Souza
 * @version 1.0
 * @since 2017
 */
package br.com.hfsframework.admin.data;

import java.util.List;

import org.apache.deltaspike.data.api.EntityRepository;
import org.apache.deltaspike.data.api.Query;
import org.apache.deltaspike.data.api.Repository;

import br.com.hfsframework.admin.model.AdmFuncionalidade;
import br.com.hfsframework.admin.model.AdmMenu;
import br.com.hfsframework.admin.model.AdmPagina;

/**
 * The Interface AdmMenuRepository.
 */
@Repository(forEntity = AdmMenu.class)
public interface AdmMenuRepository extends EntityRepository<AdmMenu, Long> {

	/**
	 * Gets the descricao.
	 *
	 * @param id
	 *            the id
	 * @return the descricao
	 */
	@Query(named = "AdmMenu.getDescricaoById")
	String getDescricao(Long id);
	
	/**
	 * Count novo.
	 *
	 * @param novo
	 *            the novo
	 * @return the long
	 */
	@Query(named = "AdmMenu.countNovo")
	Long countNovo(String novo);

	/**
	 * Count antigo novo.
	 *
	 * @param antigo
	 *            the antigo
	 * @param novo
	 *            the novo
	 * @return the long
	 */
	@Query(named = "AdmMenu.countAntigo")
	Long countAntigo(String antigo, String novo);
	
	/**
	 * Find menu raiz.
	 *
	 * @return the list
	 */
	@Query(named="AdmMenu.findMenuRaiz")
	List<AdmMenu> findMenuRaiz();
	
	/**
	 * Find menu raiz.
	 *
	 * @param nomeItemMenu
	 *            the nome item menu
	 * @return the list
	 */
	@Query(named="AdmMenu.findMenuRaizByDescricao")
	List<AdmMenu> findMenuRaiz(String nomeItemMenu);
	
	/**
	 * Count menu by funcionalidade.
	 *
	 * @param funcionalidade
	 *            the funcionalidade
	 * @return the integer
	 */
	@Query(named="AdmMenu.countMenuByFuncionalidade")
	Integer countMenuByFuncionalidade(AdmFuncionalidade funcionalidade);
	
	/**
	 * Find menus filhos.
	 *
	 * @param menu
	 *            the menu
	 * @return the list
	 */
	@Query(named="AdmMenu.findMenusFilhos")
	List<AdmMenu> findMenusFilhos(AdmMenu menu);
	
	/**
	 * Find admin menu pai by cod funcionario.
	 *
	 * @param funcionalidade
	 *            the funcionalidade
	 * @return the list
	 */
	@Query(named="AdmMenu.findAdminMenuPaiByFuncionalidade")
	List<AdmMenu> findAdminMenuPaiByCodFuncionario(AdmFuncionalidade funcionalidade);

	/**
	 * Find menu pai by cod funcionario.
	 *
	 * @param funcionalidade
	 *            the funcionalidade
	 * @return the list
	 */
	@Query(named="AdmMenu.findMenuPaiByFuncionalidade")
	List<AdmMenu> findMenuPaiByCodFuncionario(AdmFuncionalidade funcionalidade);
	
	/**
	 * Find pagina by menu.
	 *
	 * @param funcionalidade
	 *            the funcionalidade
	 * @param idMenu
	 *            the id menu
	 * @return the adm pagina
	 */
	@Query(named="AdmMenu.findPaginaByMenu")
	AdmPagina findPaginaByMenu(AdmFuncionalidade funcionalidade, Long idMenu);
	
	/*
	 * Find funcionarios por menu.
	 *
	 * @param menu
	 *            the menu
	 * @return the list
	 */
	//@Query(named = "AdmMenu.findFuncionariosPorMenu")
	//List<AdmFuncionarioDTO> findFuncionariosPorMenu(AdmMenu menu);
	
}

File: csharp-hfsframework/hfsframework-admin/hfsframework.admin.data/AdmMenuRepository.cs
using System.Collections.Generic;
using hfsframework.hfsbase;
using hfsframework.admin.model;
using hfsframework.util;
using NHibernate;

namespace hfsframework.admin.data
{
    public class AdmMenuRepository: EntityRepository<AdmMenu, long>
    {
        public override bool existeNovo(string novo)
        {
            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmMenu.countNovo");
                q.SetString(0, novo.ToLower());
                return (q.UniqueResult<long>() > 0);
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

        public override bool existeAntigo(long id, string novo)
        {
            string antigo = getDescricao(id);

            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmMenu.countAntigo");
                q.SetString(0, antigo.ToLower());
                q.SetString(1, novo.ToLower());
                return (q.UniqueResult<long>() > 0);
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

        public override string getDescricao(long id)
        {
            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmMenu.getDescricaoById");
                q.SetInt64(0, long.Parse(id.ToString()));
                return q.UniqueResult<string>();
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

        public IList<AdmMenu> findMenuRaiz()
        {
            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmMenu.findMenuRaiz");
                return q.List<AdmMenu>();
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

        public IList<AdmMenu> findMenuRaiz(string nomeItemMenu)
        {
            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmMenu.findMenuRaizByDescricao");
                q.SetString(0, nomeItemMenu);
                return q.List<AdmMenu>();
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

        public int countMenuByFuncionalidade(AdmFuncionalidade funcionalidade)
        {
            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmMenu.countMenuByFuncionalidade");
                q.SetEntity(0, funcionalidade);
                return q.UniqueResult<int>();
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

        public IList<AdmMenu> findMenusFilhos(AdmMenu menu)
        {
            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmMenu.findMenusFilhos");
                q.SetEntity(0, menu);
                return q.List<AdmMenu>();
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

        public IList<AdmMenu> findAdminMenuPaiByCodFuncionario(AdmFuncionalidade funcionalidade)
        {
            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmMenu.findAdminMenuPaiByFuncionalidade");
                q.SetEntity(0, funcionalidade);
                return q.List<AdmMenu>();
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

        public IList<AdmMenu> findMenuPaiByCodFuncionario(AdmFuncionalidade funcionalidade)
        {
            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmMenu.findMenuPaiByFuncionalidade");
                q.SetEntity(0, funcionalidade);
                return q.List<AdmMenu>();
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

        public AdmPagina findPaginaByMenu(AdmFuncionalidade funcionalidade, long idMenu)
        {
            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmMenu.findPaginaByMenu");
                q.SetEntity(0, funcionalidade);
                q.SetInt64(1, idMenu);
                return q.UniqueResult<AdmPagina>();
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

    }
}
